commands/credentials.js

```diff
@@ -11,12 +13,57 @@ function * run (context, heroku) {
   let showCredentials = co.wrap(function * () {
     const host = require('../lib/host')
     let addon = yield fetcher.addon(app, args.database)
+    let attachments = []
+
+    function formatAttachment (attachment) {
+      let attName = cli.color.addon(attachment.name)
+
+      let output = [cli.color.dim('as'), attName]
+      let appInfo = `on ${cli.color.app(attachment.app.name)} app`
+      output.push(cli.color.dim(appInfo))
+
+      return output.join(' ')
+    }
+
+    function renderAttachment (attachment, app, isFirst) {
+      let line = isFirst ? '└─' : '├─'
+      let attName = formatAttachment(attachment)
+      return printf(' %s %s', cli.color.dim(line), attName)
+    }
+
+    function presentCredential (cred) {
+      let credAttachments = []
+      if (cred !== 'default') {
+        credAttachments = attachments.filter(a => a.namespace === `credential:${cred}`)
+      } else {
+        credAttachments = attachments.filter(a => a.namespace === null)
+      }
+      let isForeignApp = (attOrAddon) => attOrAddon.app.name !== app
+      let atts = sortBy(credAttachments,
+        isForeignApp,
+        'name',
+        'app.name'
+      )
+
+      // render each attachment under the credential
+      let attLines = atts.map(function (attachment, idx) {
+        let isFirst = (idx === credAttachments.length - 1)
+        return renderAttachment(attachment, app, isFirst)
+      })
+
+      return [cred].concat(attLines).join('\n')
+    }
```

Comment:
This seems pretty fiddly. The code used by `addons` to do basically the same thing is not reusable here?